tests/components/tellduslive/test_config_flow.py

```diff
@@ -0,0 +1,230 @@
+# flake8: noqa pylint: skip-file
+"""Tests for the TelldusLive config flow."""
+import asyncio
+from unittest.mock import Mock, patch
+
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.components.tellduslive import (
+    APPLICATION_NAME, DOMAIN, KEY_HOST, KEY_SCAN_INTERVAL, SCAN_INTERVAL,
+    config_flow)
+
+from tests.common import MockConfigEntry, MockDependency, mock_coro
+
+
+def init_config_flow(hass, side_effect=None):
+    """Init a configuration flow."""
+    flow = config_flow.FlowHandler()
+    flow.hass = hass
+    flow._get_auth_url = Mock(
+        return_value=mock_coro('https://example.com'),
+        side_effect=side_effect)
+    return flow
+
+
+@pytest.fixture
+def supports_local_api():
+    """Set TelldusLive supports_local_api."""
+    return True
+
+
+@pytest.fixture
+def authorize():
+    """Set TelldusLive authorize."""
+    return True
+
+
+@pytest.fixture
+def mock_tellduslive(supports_local_api, authorize):
+    """Mock tellduslive."""
+    with MockDependency('tellduslive') as mock_tellduslive_:
+        mock_tellduslive_.supports_local_api.return_value = supports_local_api
+        mock_tellduslive_.Session().authorize.return_value = authorize
+        mock_tellduslive_.Session().access_token = 'token'
+        mock_tellduslive_.Session().access_token_secret = 'token_secret'
+        yield mock_tellduslive_
+
+
+def test_auth_url(hass, mock_tellduslive):
+    """Test _get_auth_url."""
+    flow = config_flow.FlowHandler()
+    with patch.object(flow, '_session') as session:
+        session.authorize_url = 'url'
+        result = flow._get_auth_url()
```

Comment:
Instead of calling this in the test, can we change the patch like in this test in `init_config_flow` and then call the corresponding steps in the tests instead?

Or perhaps update the mocks in the fixture mock_tellduslive?

It's good if we avoid calling internal functions and just call the entry points while patching in the perifery, outside the things we want to test.